### DISCUSSION TITLE
Vitocal 200-A kompatibel ?

### DISCUSSION THREAD

**Discussion by aw1604** [open3e/open3e#84] [General]
Hallo Zusammen,

Richtig tolles Projekt ! Gibt es eine Übersicht mit welchen Typen das funktioniert bzw. schon erfolgreich umgesetzt wurde ? Ich habe eine Vitocal-200A im Einsatz, und die ViCare-App / HA vicare Integration funktioniert zwar, ist vom Datenumfang aber eher enttäuschend. 

LG, Andreas

**Comment by philippoo66** [comment]
Moin Andreas!

Grundsätzlich ist jedes Gerät der E3 Generation kompatibel mit open3e (die Brennstoffzellen sind in ein paar Punkten etwas 'speziell'...). Das hilft dir jetzt auch nicht weiter, aber wenn ich Viessmann's Prospekt-Begriff "_One Base_" sage, klingelt es wahrscheinlich.

Diese neue Generation hat das WLAN Interface integriert, man braucht also **kein Vitoconnect** mehr um die Vicare App zu nutzen und es gibt die alte Optolink Schnittstelle **nicht** mehr.

![grafik](https://github.com/open3e/open3e/assets/122479122/9f3e588c-f974-41ae-aed1-b21eabc9ee0b)

![grafik](https://github.com/open3e/open3e/assets/122479122/c9047a95-a094-405c-80f8-947380794cdb)

im Zweifel musst du die (ersten 7 Ziffern der) Herstellernummer posten, dann bekommt man meist raus, was das für ein Gerät ist. Die meisten Vitocals sind 'One Base', es gibt aber auch noch welche ohne CAN Bus.

Grüsse!
Phil

**Comment by aw1604** [comment]
Danke ! Leider sieht es so aus, als ob meine Vitocal 200-A nirgends einen CAN-Bus stecker hat, bleibt mir nur auf das ältere vcontrold zu setzen :)

**Reply by philippoo66** [reply]
ich hab da noch n paar schöne Sachen für Optolink (einfache Python Skripts, Windoofs Software), du brauchst da nicht das von-hinten-durch-die-Brust-ins-Auge vcontrold!

**Comment by aw1604** [comment]
Danke :) Also ich möchte ja nur:
1. Den Vitoconnect-Kram beibehalten (wegen der Garantieverlängerung) und 
2. meine Daten in mein Home-Assistant bekommen

Aktuell plane ich eine USB-Bridge auf Raspi-Basis / python-usb-bridge script, das die Kommunikation zwischen Optolink und WP durchreicht und mir einen tcp-socket zum mitlesen und eingreifen bietet. Ob ich daran einen vcontrold oder etwas anderes anflansche ist mir egal, hauptsache ich kriege meine Daten z.b. per mqtt ins Home-Assistant weitergereicht. Wenn Deine scripts mit einem tcp-socket arbeiten und eleganter/einfacher als vcontrold sind dann gerne.. (aber Windoofs geht gar nicht!)

Ach ja - meine Vitocal 200-A identifiziert sich gerne als CU401B_S, und hat auf der Platine einen Modbus, aber keinen CAN-Bus Stecker.

LG, Andreas

**Reply by philippoo66** [reply]
python-usb-bridge script - an sowas bastele ich grade. Blöd ist dass die Initialisierug des 300er Protokolls und das Protokoll selber unterschiedliche 'Formate' haben, das macht es etwas schwieriger. Aktuell warte ich, bis eine gewisse Zeit keine neuen Daten mehr kommen, um das Ende einer 'Sendung' zu identifizieren, aber das ist noch etwas instabil und macht die Sache natürlich noch langsamer...

hast du schon einen Ansatz?

Es gibt ja schon eine [Bridge](https://github.com/h0nIg/vcontrol-vitoconnect-bridge), aber die funktioniert irgendwie nicht. Ich kann da nirgends eine geordnete Folge erkennen, die schreibt einfach alles weiter und guckt garnicht, ob eine Sache zuende ist. und die macht wieder den Umweg über vcontrold...

Wenn man eine ordentliche 'Weiche' hat, ist es ja fast egal, was man wohin weiter reicht. Den MQTT Part aus open3e zu übernehmen wäre einfach, Markus 'sarnau' hatte ja vor Ewigkeiten auch schon mal was geschrieben, aber auch mit vcontrol...

Grüsse!
Phil
